addons/mail/static/src/utils/common/format.js

```diff
@@ -268,4 +268,21 @@ export function parseEmail(text) {
     return [text, false];
 }
 
+/**
+ * splits the string using the splitChar and find all the invalid emails from it.
+ *
+ * @param {string} splitPattern
+ * @param {string} emailStr
+ * @return {object}
+ */
+export function findInvalidEmailFromText(splitPattern, emailStr){
+    const emailList = emailStr.split(splitPattern);
+    const invalidEmails = emailList.filter(email => email !== '' && !parseEmail(email.trim())[1]);
+    const emailInfo = {
+        'invalidEmails': invalidEmails,
+        'emailList': emailList,
+    }
+    return emailInfo
+}
+
```

Comment:
This is very specific function that is only used in `website_event`, it should not be put in `mail/` as it is.

Reply:
This function is called by the website_event and by the appointment addon two times.

I thought the work done here was enough generic to be put in the mail addons. Otherwise, I would have to duplicate the function to the website_event...

Reply:
It's definitely not generic enough. The internal implementation has nothing to do with email. The only part that gives the wrong confidence it has to do with email is payload returning "email" in keys, which is just arbitrary. It's also very opinionated what it means "invalid" email.

Reply:
@Clement-Cardot Maybe link enterprise PR in PR message so that it gives an easy access to the other branches :) if it is used two times it can be added in mail as we have no other common module between those two.

Note that naming is a bit misleading: it find valid and invalid emails, not just invalid ^^ . Maybe ``checkEmailValidity`` ?

Reply:
PR links updated :)
Yep indeed the function name can be change ^^
So I keep the MOV commit for now ?